addons/controllers/clusterbootstrap_controller.go

```diff
@@ -785,21 +913,30 @@ func patchSecretWithLabels(secret *corev1.Secret, pkgName, clusterName string) b
 func (r *ClusterBootstrapReconciler) cloneSecretsAndProviders(cluster *clusterapiv1beta1.Cluster, bootstrap *runtanzuv1alpha3.ClusterBootstrap,
 	templateNS string, log logr.Logger) ([]*corev1.Secret, []*unstructured.Unstructured, error) {
 
-	var createdProviders []*unstructured.Unstructured
-	var createdSecrets []*corev1.Secret
-
 	packages := append([]*runtanzuv1alpha3.ClusterBootstrapPackage{
 		bootstrap.Spec.CPI,
 		bootstrap.Spec.CSI,
 		bootstrap.Spec.Kapp,
 	}, bootstrap.Spec.CNIs...)
 	packages = append(packages, bootstrap.Spec.AdditionalPackages...)
 
+	return r.cloneSecretsAndProvidersFromList(cluster, packages, templateNS, log)
```

Comment:
Could L320 get refactored to add the following and then also call ```cloneSecretsAndProvidersFromList()``` at line 320? It is a bit confusing to have both ```cloneSecretsAndProviders``` and ```cloneSecretsAndProvidersFromList```, and with this change ```cloneSecretsAndProviders``` would be very small function and doesn't need to be a separate function of its own
```

	packages := append([]*runtanzuv1alpha3.ClusterBootstrapPackage{
		bootstrap.Spec.CPI,
		bootstrap.Spec.CSI,
		bootstrap.Spec.Kapp,
	}, bootstrap.Spec.CNIs...)
	packages = append(packages, bootstrap.Spec.AdditionalPackages...)
```

Reply:
Sounds like a plan

Reply:
+1 combine both and caller explicitly provides the packages whose secrets and providers need to be cloned.